invenio/modules/documents/utils.py

```diff
@@ -88,12 +88,23 @@ def _get_legacy_bibdoc(recid, filename=None):
     """
     from invenio.ext.login import current_user
     from invenio.legacy.bibdocfile.api import BibRecDocs
-    paths = [
+    return [
         (bibdoc.fullpath, bibdoc.is_restricted(current_user))
         for bibdoc in BibRecDocs(recid).list_latest_files(list_hidden=False)
         if not bibdoc.subformat and not filename or
         bibdoc.name + bibdoc.superformat == filename
```

Comment:
Sorry seeing this for the first time: How can one retrieve a file having a subformat? e.g. 
`foo.pdf;pdfa`? Note also that what you obtain from `list_latest_files` are `BibDocFile` objects not `BibDoc` objects.

Can this filter be changed e.g. to:
`if not filename or bibdoc.get_fullname() == fullname`?